packages/react/src/components/Table/EmptyTable/EmptyTable.jsx

```diff
@@ -49,55 +55,61 @@ const EmptyTable = ({
   // TODO: remove deprecated 'testID' in v3
   testID,
   testId,
-}) => (
-  <TableBody
-    id={`${id}-empty-table`}
-    // TODO: remove deprecated 'testID' in v3
-    data-testid={testID || testId}
-  >
-    <TableRow className={`${iotPrefix}--empty-table--table-row`}>
-      <TableCell colSpan={totalColumns}>
-        {React.isValidElement(emptyState) ? (
-          emptyState
-        ) : (
-          <div className="empty-table-cell--default">
-            {isFiltered ? (
-              <EmptyState
-                icon="no-result"
-                title={messageWithFilters}
-                body={messageWithFiltersBody || ''}
-                action={
-                  onEmptyStateAction
-                    ? {
-                        label: buttonLabelWithFilters,
-                        onClick: onEmptyStateAction,
-                        kind: 'secondary',
-                      }
-                    : null
-                }
-              />
-            ) : (
-              <EmptyState
-                icon="empty"
-                title={message}
-                body={messageBody || ''}
-                action={
-                  onEmptyStateAction
-                    ? {
-                        label: buttonLabel,
-                        onClick: onEmptyStateAction,
-                      }
-                    : null
-                }
-              />
-            )}
-          </div>
-        )}
-      </TableCell>
-    </TableRow>
-  </TableBody>
-);
+  noDataIcon,
+}) => {
+  const isFilterEmptyTable = (
+    <EmptyState
+      icon="no-result"
+      title={messageWithFilters}
+      body={messageWithFiltersBody || ''}
+      action={
+        onEmptyStateAction
+          ? {
+              label: buttonLabelWithFilters,
+              onClick: onEmptyStateAction,
+              kind: 'secondary',
+            }
+          : null
+      }
+    />
+  );
 
+  const defaultEmptyTable = (
+    <EmptyState
+      icon={noDataIcon !== '' ? noDataIcon : 'empty'}
```

Comment:
empty string are "falsy" so you can do this:
```suggestion
      icon={noDataIcon ?? 'empty'}
```